### PR TITLE
Added z-Index based render order

### LINKED ISSUE #81: Add flexible render order
The render order is currently defined by a hard-coded order of list:
https://github.com/VEGO-Engine/Engine/blob/e493960fe0c1107662df6446f2f8a12cae27f681/src/GameInternal.cpp#L169-L188
This is sub-optimal, as it restricts the game dev to use our defined groups for objects, and they are always in the same fixed order. A better approach would be to either allow arbitrary orders, perhaps using a z-index.

### PR DESCRIPTION
**Pull request by Nanogamer7**
Requires update of chicken game, pushed as https://github.com/VEGO-Engine/chickengame/tree/zindex

### CODE REVIEW

File: src/TileComponent.cpp
Comment (Nanogamer7) [inline src/TileComponent.cpp:1]: Change redundant after #53

File: src/TransformComponent.cpp
Comment (Nanogamer7) [inline src/TransformComponent.cpp:1]: Issue when compiling with clang

File: src/RenderManager.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include "RenderObject.h"
+#include <algorithm>
+
+
+void RenderManager::renderAll()
+{
+	if (!this->isSorted) {
+		std::ranges::sort(this->renderObjects, RenderObject::ZIndexComparator());
+	}
+	std::ranges::for_each(this->renderObjects, [](RenderObject* obj) {
```
Comment (Freezarite) [inline src/RenderManager.cpp:11]: unnecessary use of for_each resulting in the import of <algorithm>. maybe change to standard for loop

Comment (Freezarite) [inline src/RenderManager.cpp]: unnecessary use of for_each resulting in the import of `<algorithm>` maybe change to standard for loop

**Review by Freezarite — CHANGES_REQUESTED** [review]

File: src/RenderManager.cpp (hunk 1 above)
Comment (Nanogamer7) [reply src/RenderManager.cpp]: Has been changed for readability, however the `<algorithm>`-library is still required for `std::sort` and `std::remove`.

**Review by Freezarite — APPROVED** [review]
lgtm

**Pull request merged**